frontend/src/modules/entitydetails/components/Tools.js

```diff
@@ -38,6 +39,23 @@ type Props = {|
  * Shows the metadata of the entity and an editor for translations.
  */
 export default class Tools extends React.Component<Props> {
+    preferredCount() {
+        const { otherlocales, user } = this.props;
+
+        if (!user.isAuthenticated) {
+            return 0;
+        }
+
+        const preferred = otherlocales.translations.reduce((count, item) => {
+            if (user.preferredLocales.indexOf(item.code) > -1) {
+                return count + 1;
+            }
+            return count;
+        }, 0);
+
+        return preferred;
+    }
```

Comment:
I think this should actually be a selector. This way, we won't recompute it every time there's a render, but only when the depending data changes.